scheduler/feasible.go

```diff
@@ -480,7 +480,8 @@ func checkConstraint(ctx Context, operand string, lVal, rVal interface{}) bool {
 
 // checkAffinity checks if a specific affinity is satisfied
 func checkAffinity(ctx Context, operand string, lVal, rVal interface{}) bool {
-	return checkConstraint(ctx, operand, lVal, rVal)
+	// We pass ok here for both values, because matchesAffinity prevalidates these
```

Comment:
@preetapan I'm not too familiar with the affinity system yet, but do we want to support similar is_set mechanics there too? - Right now, affinities are probably inheriting the new != matching behaviour?

Reply:
yup, they share the same set of operators (except for distincthost/distinctproperty) and the underlying matching code so this should just work for affinities too.

Reply:
since affinities are optional/best effort with weights it didn't make sense to me to support lok/rok values. ie creating two affinities one for the value being set and another for the value matching a value won't combine well when calculating a final score. So I wanted to preserve the existing implementation (only match if both the values are set)